### DISCUSSION TITLE
Quering and retriving nested objects

### DISCUSSION THREAD

**Discussion by InverMN** [ravendb/ravendb#16199] [Q&A] [answered]
Hi guys,

I have a dead simple problem, that really starts setting me up. 

So... in the simplified version I have one read model:

```csharp
class Customer {
    public string Id { get; set; }
    public PersonalData PersonalInformation { get; set; }
}

class PersonalData {
    public string FirstName { get; set; }
    public string LastName { get; set; }
}
```

And I am quering the container with many records like this. The problem is that response contains PersonalData fields with all null values.

I was trying to use .Include(c => c.PersonalInformation), but seems like it doesn't work as I excepted.

Any solution for my "problem"? I just can't find API to do that.

**Comment by georgiosd** [comment]
Can you show your code?

**Reply by InverMN** [reply]
Yeah, here catch some cut-out of code:

```csharp
public class WorkCenterDetails
{
    public string Id { get; set; }
    public string Symbol { get; set; }
    public string Name { get; set; }
    public WorkCenterLocalization Location { get; set; }
    public EnumWorkCenterStatus Status { get; set; }
    public string ResourceId { get; set; }
}
    
public class WorkCenterLocalization {
    public string OrganizationId { get; }
    public string FactoryId { get; }
    public string BuildingId { get; }
    public string FloorId { get; }
    public string ZoneId { get; }
}

public static class Queries
{
    public static Task<List<ReadModels.WorkCenterDetails>> Query(this IAsyncDocumentSession session, QueryModels.GetAllWorkCenters query) 
        => session
            .Query<ReadModels.WorkCenterDetails>()
            .Include(w => w.Location)
            .Skip((query.Page -1) * query.PageSize)
            .Take(query.PageSize)
            .ToListAsync()
            .MapFromRavenDb();
}
```

There's much more glue code, but by debugging I figured out Query() returns nulls on Location. (MapFromRavenDb gets nulled Locations in arguments already)

**Comment by georgiosd** [comment, marked as answer]
You need to use As<T>() or ProjectInto<T>() to tell Raven to get those
fields from the original document.

A different strategy would be to store the fields in the index, but that's
not recommended for this use case.

Also, Include() only works for properties that are ids.

Hopefully I've been given you enough pointers to navigate the docs :)

**Reply by InverMN** [reply]
Okay, I used the methods, but what was kinda surprise for me, the problem was in my `WorkCenterLocalization` class definition. Properties had only getters but setters. RavenDB client just couldn't set values.

But I have resolved the problem, 

Thanks for the Help!

**Reply by georgiosd** [reply]
It has nothing to do with the getters/setters. The information doesn't exist in the index, unless you store it.

**Reply by InverMN** [reply]
anyway, it has started working
